lib/logstash/inputs/syslog.rb

```diff
@@ -59,8 +62,10 @@ def register
       "tag_on_failure" => ["_grokparsefailure_sysloginput"],
     )
 
+    locale = @config["locale"][0] if @config["locale"] != nil and @config["locale"][0] != nil
```

Comment:
Hi @nwknu, we should remove this check as @colinsurprenant pointed out and directly pass it to the date filter if it is not nil or empty